lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/InitMavenRequestTest.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat Inc. and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.lemminx.extensions.maven;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.util.Properties;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(NoMavenCentralIndexExtension.class)
+public class InitMavenRequestTest {
+
+	private static final String TEST_LOCAL_REPO_PATH = "/.m2/bug383repository";
+	private static final String TEST_DUMMY_PROPERTY_NAME = "DUMMY_USER_HOME";
+	private static final String MAVEN_LOCAL_REPO_PROPERTY_NAME = "maven.repo.local";
+	private static final String USER_HOME_PROPERTY_NAME = "user.home";
+
+	@Test
+	public void testUserSettingsWithEnvVariables() throws Exception {
+		final File localSettingsFile = new File(getClass().getResource("/settings-local-repo-test.xml").toURI());
+		String localSettingsContent = FileUtils.readFileToString(localSettingsFile, "UTF-8");
+		String localRepoProperty = System.getProperty(MAVEN_LOCAL_REPO_PROPERTY_NAME);
+		Properties properties = System.getProperties();
+		try {
+			properties.remove(MAVEN_LOCAL_REPO_PROPERTY_NAME); // Not to use local repository set from properties
+			
+			// Add a DUMMY environment variable to check later if it's really resolved to a real path
+			String userHomeProperty = properties.getProperty(USER_HOME_PROPERTY_NAME);
+			assertNotNull(userHomeProperty);
+			properties.put(TEST_DUMMY_PROPERTY_NAME, userHomeProperty);
+			System.setProperties(properties);
+			
+			MavenLemminxExtension plugin = new MavenLemminxExtension();
+			plugin.settings.setGlobalSettings(null);
+			plugin.settings.setUserSettings(localSettingsContent);
```

Comment:
A file is expected for settings (doc is missing), so it should be
```suggestion
			plugin.settings.setUserSettings(localSettingsFile.getAbsolutePath());
```

Reply:
Hm. It doesn't work like this....
```
org.opentest4j.AssertionFailedError: expected: </home/user/.m2/bug383repository> but was: </home/user/projects/lemminx/lemminx-maven/lemminx-maven/${env.DUMMY_USER_HOME}/.m2/bug383repository>
	at org.eclipse.lemminx.extensions.maven.InitMavenRequestTest.testUserSettingsWithEnvVariables(InitMavenRequestTest.java:51)
```

Reply:
That's the same issue I see, and the same one on CI. So overall it looks like it's a good thing you can reproduce it as well.

Reply:
Actually, I see that you're right and the absolute config path is to be used here.
Now I have two problems.... 1) The system properties aren't get resolved in paths (that DUMMY property I have added) - trying to find out why.
2) The second test case added for env. variable added (with the value of maven project location) goes successfully (the var is resolved), but fails on that repository is not accessible - I don;t understand why (is that directory is kinda read only for test or there was some problem on create - not exceptions logged though)

Reply:
Nevermind. I have found an issue in my code, so the system/env. properties was not taking into account on interpolate() calls. Now fixed.